core/src/main/java/org/mule/runtime/core/internal/config/bootstrap/AbstractRegistryBootstrap.java

```diff
@@ -264,7 +281,7 @@ private void registerObjects(List<ObjectBootstrapProperty> bootstrapProperties)
 
   private void registerObject(ObjectBootstrapProperty bootstrapProperty) throws Exception {
     try {
-      if (!bootstrapProperty.getArtifactType().equals(ALL) && !bootstrapProperty.getArtifactType().equals(artifactType)) {
+      if (!bootstrapProperty.getArtifactTypes().contains(artifactType) && !bootstrapProperty.getArtifactTypes().contains(ALL)) {
```

Comment:
Is still required to compare against ALL? I think that ALL should not be used at all